tests/test_orient_symmetric.py

```diff
@@ -157,32 +150,49 @@ def testRelativeViewingDirections(self, order):
         # Estimate relative viewing directions.
         vijs, viis = cl_symm._estimate_relative_viewing_directions_c3_c4()
 
-        # Calculate the mean squared error for vijs.
-        errs_vijs = np.zeros(n_pairs)
-        diffs_vijs = np.zeros(2)
-        for idx, (vij_gt, vij) in enumerate(zip(vijs_gt, vijs)):
-            diffs_vijs[0] = norm(vij - vij_gt)
-            diffs_vijs[1] = norm(J_conjugate(vij) - vij_gt)
-            errs_vijs[idx] = np.min(diffs_vijs)
-        mse_vijs = np.mean(errs_vijs**2)
-
-        # Calculate the mean squared error for viis.
-        errs_viis = np.zeros(n_img)
-        diffs_viis = np.zeros(2)
-        for idx, (vii_gt, vii) in enumerate(zip(viis_gt, viis)):
-            diffs_viis[0] = norm(vii - vii_gt)
-            diffs_viis[1] = norm(J_conjugate(vii) - vii_gt)
-            errs_viis[idx] = np.min(diffs_viis)
-        mse_viis = np.mean(errs_viis**2)
-
-        # Check that MSE is small.
-        # MSE is better for C3 than C4.
-        if order == 3:
-            self.assertTrue(mse_vijs < 0.0008)
-            self.assertTrue(mse_viis < 0.0002)
-        else:
-            self.assertTrue(mse_vijs < 0.01)
-            self.assertTrue(mse_viis < 0.0011)
+        # Since ground truth vijs and viis are rank 1 matrices they span a 1D subspace.
+        # We use SVD to find this subspace for our estimates and the ground truth relative viewing directions.
+        # We then calculate the angular distance between these subspaces (and take the mean).
+        # SVD's:
+        uij_gt, _, _ = np.linalg.svd(vijs_gt)
+        uii_gt, _, _ = np.linalg.svd(viis_gt)
+        uij_est, _, _ = np.linalg.svd(vijs)
+        uii_est, _, _ = np.linalg.svd(viis)
+        uij_J_est, _, _ = np.linalg.svd(J_conjugate(vijs))
+        uii_J_est, _, _ = np.linalg.svd(J_conjugate(viis))
+
+        # Ground truth 1D supbspaces.
+        uij_gt = uij_gt[:, :, 0]
+        uii_gt = uii_gt[:, :, 0]
+
+        # 1D subspace of estimates.
```

Comment:
Why take the 1D subspace and not the entire thing? If the estimated `uij` deviate from a rank-one matrix, that's a problem, no?

Reply:
As discussed, I added a check to make sure that the estimates are sharply approximate to rank 1.

Reply:
Good.